src/interfaces.ts

```diff
@@ -31,9 +31,9 @@ export interface Split {
 
 export interface Allocation {
   key: string;
-  rules: Rule[];
-  startAt?: Date;
-  endAt?: Date;
+  rules?: Rule[];
+  startAt?: string; // ISO 8601
+  endAt?: string; // ISO 8601
```

Comment:
This is annoying, but I see why we need to do this as JSON doesn't support dates.

Reply:
In the Python SDK, Pydantic takes care of the conversion for us. We can try to do something similar (converting to dates when parsing config) at a later time.